packages/now-cli/src/util/dev/server.ts

```diff
@@ -507,23 +497,24 @@ export default class DevServer {
     return {};
   }
 
-  async getNowConfig(canUseCache: boolean = true): Promise<NowConfig> {
+  clearNowConfigPromise = () => {
+    this.getNowConfigPromise = null;
+  };
+
+  getNowConfig(): Promise<NowConfig> {
     if (this.getNowConfigPromise) {
       return this.getNowConfigPromise;
     }
-    this.getNowConfigPromise = this._getNowConfig(canUseCache);
-    try {
-      return await this.getNowConfigPromise;
-    } finally {
-      this.getNowConfigPromise = null;
-    }
-  }
+    this.getNowConfigPromise = this._getNowConfig();
 
-  async _getNowConfig(canUseCache: boolean = true): Promise<NowConfig> {
-    if (canUseCache && this.cachedNowConfig) {
-      return this.cachedNowConfig;
-    }
+    // Clean up the promise once it has resolved
+    const clear = this.clearNowConfigPromise;
+    this.getNowConfigPromise.then(clear, clear);
```

Comment:
TIL about the second parameter to `then()`. I have been using `then().catch()` all these years 🤯

Reply:
Is that different than `.finally()`?

Reply:
I always forget about it because it's relatively new (Node 10+), but `engines.node` is already set to 10 for CLI, so we could use `.finally()` here.

```suggestion
    this.getNowConfigPromise.finally(clear);
```